src/Compilers/VisualBasic/Portable/Symbols/Source/SourceNamedTypeSymbol.vb

```diff
@@ -2531,6 +2533,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                     AddSynthesizedAttribute(attributes, compilation.SynthesizeTupleNamesAttribute(baseType))
                 End If
             End If
+
+            ' Add MetadataUpdateOriginalTypeAttribute when a reloadable type is emitted to EnC delta
+            If moduleBuilder.EncSymbolChanges?.IsReplaced(CType(Me, ISymbolInternal).GetISymbol()) = True Then
+                ' Note that we use this source named type symbol in the attribute argument (of System.Type).
+                ' We do not have access to the original symbol from this compilation. However, System.Type
+                ' is encoded in the attribute as a string containing a fully qualified type name.
+                ' The name of the current type symbol as provided by ISymbol.Name is the same as the name of
+                ' the original type symbol that is being replaced by this type symbol.
+                ' The "#{generation}" suffix is appended to the TypeDef name in the metadata writer,
+                ' but not to the attribute value.
+                Dim originalType = Me
+
+                AddSynthesizedAttribute(
+                    attributes,
+                    compilation.TrySynthesizeAttribute(
+                        WellKnownMember.System_Runtime_CompilerServices_MetadataUpdateOriginalTypeAttribute__ctor,
+                        ImmutableArray.Create(New TypedConstant(compilation.GetWellKnownType(WellKnownType.System_Type), TypedConstantKind.Type, originalType)),
+                        isOptionalUse:=True))
```

Comment:
The target .NET version might not have the attribute defined (the attribute is available since .NET 7). The absence of the attribute on previous version does not break anything because there is no code that would use the attribute there.